histoire w/ unocss and vue throws error on dev/build

### Describe the bug

When trying to add histoire to showcase components using unocss, it runs into the following error

```
[unocss-hmr] TypeError: __vite_ssr_import_meta__.hot.send is not a function
    at C:\__uno.css:17:40
    at async ViteNodeRunner.directRequest (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.28.4/node_modules/vite-node/dist/client.mjs:312:5)
    at async ViteNodeRunner.cachedRequest (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.28.4/node_modules/vite-node/dist/client.mjs:156:14)
    at async ViteNodeRunner.dependencyRequest (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.28.4/node_modules/vite-node/dist/client.mjs:204:14)
    at async C:\DEV\uno-x-histoire\histoire.setup.ts:1:31
    at async ViteNodeRunner.directRequest (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.28.4/node_modules/vite-node/dist/client.mjs:312:5)
    at async ViteNodeRunner.cachedRequest (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.28.4/node_modules/vite-node/dist/client.mjs:156:14)
    at async ViteNodeRunner.dependencyRequest (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.28.4/node_modules/vite-node/dist/client.mjs:204:14)
    at async C:\DEV\uno-x-histoire\node_modules\.pnpm\@histoire+plugin-vue@0.16.1_histoire@0.16.2_vite@4.3.9_vue@3.2.47\node_modules\@histoire\plugin-vue\dist\bundled\client\server\run.js:2:31
    at async ViteNodeRunner.directRequest (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.28.4/node_modules/vite-node/dist/client.mjs:312:5)
src\components\HelloWorld.story.vue 407ms (setup:63ms, execute:36ms, run:308ms)
```

I looked a bit and [this commit](https://github.com/vitest-dev/vitest/commit/b1624db5d3baf26f7e1c69b23cf4c1335562ccde) shouldve fixed the issue from `vite-node` throwing this. 
But even when override `vite-node` to `^0.32.0`, i then get this error: 

```
Error while collecting story C:/DEV/uno-x-histoire/src/components/HelloWorld.story.vue:
Error: Cannot find module '/__resolved__virtual:$histoire-support-plugins-collect' imported from '/@fs/C:/DEV/uno-x-histoire/node_modules/.pnpm/histoire@0.16.2_vite@4.3.9/node_modules/histoire/dist/node/collect/run.js'.

- If you rely on tsconfig.json to resolve modules, please install "vite-tsconfig-paths" plugin to handle module resolution.
 - Make sure you don't have relative aliases in your Vitest config. Use absolute paths instead. Read more: https://vitest.dev/guide/common-errors
    at ViteNodeRunner._resolveUrl (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.32.0/node_modules/vite-node/dist/client.mjs:211:21)
    at async ViteNodeRunner.resolveUrl (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.32.0/node_modules/vite-node/dist/client.mjs:228:14)
    at async request (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.32.0/node_modules/vite-node/dist/client.mjs:243:32)  
    at async C:\DEV\uno-x-histoire\node_modules\.pnpm\histoire@0.16.2_vite@4.3.9\node_modules\histoire\dist\node\collect\run.js:1:31    
    at async ViteNodeRunner.directRequest (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.32.0/node_modules/vite-node/dist/client.mjs:345:5)
    at async ViteNodeRunner.cachedRequest (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.32.0/node_modules/vite-node/dist/client.mjs:190:14)
    at async ViteNodeRunner.executeFile (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/vite-node@0.32.0/node_modules/vite-node/dist/client.mjs:162:12)
    at async default (file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/histoire@0.16.2_vite@4.3.9/node_modules/histoire/dist/node/collect/worker.js:41:22)
    at async file:///C:/DEV/uno-x-histoire/node_modules/.pnpm/@akryum+tinypool@0.3.1/node_modules/@akryum/tinypool/dist/esm/worker.js:111:20
```

### Reproduction

https://github.com/prazdevs/histoire-unocss-bug

### System Info

```shell
System:
    OS: Windows 10 10.0.19044
    CPU: (16) x64 AMD Ryzen 7 3700X 8-Core Processor
    Memory: 6.62 GB / 15.95 GB
  Binaries:
    Node: 18.15.0 - ~\AppData\Local\Volta\tools\image\node\18.15.0\node.EXE    
    Yarn: 1.22.19 - ~\AppData\Local\Volta\tools\image\yarn\1.22.19\bin\yarn.CMD
    npm: 9.5.0 - ~\AppData\Local\Volta\tools\image\node\18.15.0\npm.CMD        
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (114.0.1823.43)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    @histoire/plugin-vue: ^0.16.1 => 0.16.1
    @vitejs/plugin-vue: ^4.1.0 => 4.1.0
    histoire: ^0.16.2 => 0.16.2
    vite: ^4.3.9 => 4.3.9
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/histoire-dev/histoire/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/histoire-dev/histoire/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://histoire.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/histoire-dev/histoire/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/histoire-dev/histoire/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/histoire/package.json
+++ b/packages/histoire/package.json
@@ -73,7 +73,7 @@
     "sade": "^1.8.1",
     "shiki-es": "^0.2.0",
     "sirv": "^2.0.3",
-    "vite-node": "0.28.4"
+    "vite-node": "^0.34.1"
   },
   "peerDependencies": {
     "vite": "^2.9.0 || ^3.0.0 || ^4.0.0"

